plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php

```diff
@@ -1434,12 +1432,22 @@ private function log_diff( array $diff ): void {
 	 * @return void
 	 */
 	private function migrate_post_record( \WC_Abstract_Order &$order, \WC_Abstract_Order $post_order ): void {
-		$this->migrate_meta_data_from_post_order( $order, $post_order );
+		$diff                 = $this->migrate_meta_data_from_post_order( $order, $post_order );
 		$post_order_base_data = $post_order->get_base_data();
 		foreach ( $post_order_base_data as $key => $value ) {
 			$this->set_order_prop( $order, $key, $value );
 		}
 		$this->persist_updates( $order, false );
+
+		/**
+		 * Fired when an HPOS order is updated from its corresponding post record on read due to a difference in the data.
+		 *
+		 * @since 8.5.0
+		 *
+		 * @param \WC_Order $order The order object.
+		 * @param array     $diff  Difference between HPOS data and post data.
+		 */
+		do_action( 'woocommerce_hpos_post_record_migrated_on_read', $order, $diff );
```

Comment:
Similar to my comment on the other hook, when I tested this by updating a post order record via wp-cli, the hook fired two different times. There was a bit of a delay, I think because this migration method gets called by a scheduled action? The first time the hook fired, $diff contained the new meta value I had added via the wp-cli command. The second time, the diff was just an empty array. In this case, if I wanted to reduce the noise from logging this hook, I could add some logic to only write a log if the diff isn't empty, but I think the question remains, why is this firing multiple times?